subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -158,25 +183,66 @@ void lll_sync_aux_prepare_cb(struct lll_sync *lll,
 	}
 }
 
+enum sync_status lll_sync_cte_is_allowed(uint8_t cte_type_mask, uint8_t filter_policy,
+					 uint8_t rx_cte_time, uint8_t rx_cte_type)
+{
+	bool cte_ok;
+
+	if (cte_type_mask & BT_HCI_LE_PER_ADV_CREATE_SYNC_CTE_TYPE_NO_FILTERING) {
+		return SYNC_STAT_ALLOWED;
+	}
+
+	/* If there is no CTEInfo in advertising PDU, Radio will not parse the S0 byte and
+	 * CTESTATUS register will hold zeros only.
+	 * Zero value in CTETime field of CTESTATUS may be used to distinguish between PDU that
+	 * includes CTEInfo or not. Allowed range for CTETime is 2-20.
+	 */
+	if (rx_cte_time == 0 && cte_type_mask & BT_HCI_LE_PER_ADV_CREATE_SYNC_CTE_TYPE_ONLY_CTE) {
+		cte_ok = false;
+	}
+
+	if (rx_cte_time > 0) {
+		if (cte_type_mask & BT_HCI_LE_PER_ADV_CREATE_SYNC_CTE_TYPE_NO_CTE) {
```

Comment:
```suggestion
		if ((cte_type_mask & BT_HCI_LE_PER_ADV_CREATE_SYNC_CTE_TYPE_NO_CTE) != 0) {
```

Reply:
This comment applies multiple places if you want to follow the MISRA guidelines about strict typing. (14.4. in https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#main-rules)

Reply:
Updated lines where cte_type_mask if used in `if` statements.